modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionPayloads.scala

```diff
@@ -28,14 +27,14 @@ case object ConstructionPayloads {
   )
 
   object Response {
-    def fromPayloadsResult(result: PayloadsResult): Response = Response(
+    def fromPayloadsResult(result: GetPayloadsResult): Response = Response(
       result.unsignedTransaction,
       result.payloads
     )
   }
 
   @derive(eqv, show)
-  case class PayloadsResult(
+  case class GetPayloadsResult(
```

Comment:
thinking out of scope of this project - imo it sounds natural that function `getMetadata` returns `Metadata` and `getPayloads` returns `Payloads` or `List[Payload]`.
Also, the domain service should be implementation agnostic. I mean, it should not know the exact implementation, it just operates on other interfaces. That being said, it should not return `MetadataResponse` itself, as the service can be used in different ways. One of them is exposing it via REST API which we actually do end of the day.

I have another suggestion, instead of having `Request` and `Response` classes, if the Response is equally the same as, let's say, `Metadata` - let's return `Metadata` directly to the endpoint.

We've started with `Request`/`Response` convention because it was an aggregation for larger structure (see `ConstructionDerive.Response` - it doesn't return `AccountIdentifier`, but it returns `AccountIdentifier` wrapped into an object). So my suggestion is to use `Request`/`Response` wherever needed, wherever we aggregate some other types into an object, and use direct case class if the request or response is exactly what we already have mapped.

Reply:
@marcinwadon I don't understand what you're looking for with this last comment. Can we meet in the morning and go over it?